agent/src/main.rs

```diff
@@ -45,44 +36,68 @@ async fn main() -> Result<(), Box<dyn std::error::Error + Send + Sync + 'static>
     let mut tasks = Vec::new();
     let node_name = env::var("AGENT_NODE_NAME")?;
 
-    // Start server for Prometheus metrics
-    tasks.push(tokio::spawn(async move {
-        run_metrics_server().await.unwrap();
-    }));
-
-    let discovery_handler_map = Arc::new(Mutex::new(HashMap::new()));
-    let discovery_handler_map_clone = discovery_handler_map.clone();
-    let (new_discovery_handler_sender, _): (
-        broadcast::Sender<DiscoveryHandlerName>,
-        broadcast::Receiver<DiscoveryHandlerName>,
-    ) = broadcast::channel(NEW_DISCOVERY_HANDLER_CHANNEL_CAPACITY);
-    let new_discovery_handler_sender_clone = new_discovery_handler_sender.clone();
-    #[cfg(feature = "agent-full")]
-    register_embedded_discovery_handlers(discovery_handler_map_clone.clone())?;
-
-    // Start registration service for registering `DiscoveryHandlers`
-    tasks.push(tokio::spawn(async move {
-        run_registration_server(discovery_handler_map_clone, new_discovery_handler_sender)
-            .await
-            .unwrap();
-    }));
+    {
+        let kube_client = Arc::new(akri_shared::k8s::KubeImpl::new().await?);
+
+        // Start server for Prometheus metrics
+        tasks.push(tokio::spawn(async move {
+            run_metrics_server().await.unwrap();
+        }));
 
-    tasks.push(tokio::spawn(async move {
-        let slot_grace_period = Duration::from_secs(SLOT_RECONCILIATION_SLOT_GRACE_PERIOD_SECS);
-        periodic_slot_reconciliation(slot_grace_period)
+        let (device_notifier, discovery_handler_registry, conf_notifier) =
```

Comment:
Nice to have consistent abreviations
```suggestion
        let (device_notifier, discovery_handler_registry, config_notifier) =
```